diff -urN cardano-node-og/cardano-cli/cardano-cli.cabal cardano-node/cardano-cli/cardano-cli.cabal
--- cardano-node-og/cardano-cli/cardano-cli.cabal	2022-06-03 17:59:54.773513963 +0200
+++ cardano-node/cardano-cli/cardano-cli.cabal	2022-06-03 18:00:35.861674072 +0200
@@ -1,7 +1,7 @@
 cabal-version: 3.0
 
 name:                   cardano-cli
-version:                1.33.0
+version:                1.35.0
 description:            The Cardano command-line interface.
 author:                 IOHK
 maintainer:             operations@iohk.io
diff -urN cardano-node-og/cardano-node/cardano-node.cabal cardano-node/cardano-node/cardano-node.cabal
--- cardano-node-og/cardano-node/cardano-node.cabal	2022-06-03 17:59:54.776513975 +0200
+++ cardano-node/cardano-node/cardano-node.cabal	2022-06-03 18:00:45.179710383 +0200
@@ -1,7 +1,7 @@
 cabal-version: 3.0
 
 name:                   cardano-node
-version:                1.33.0
+version:                1.35.0
 description:            The cardano full node
 author:                 IOHK
 maintainer:             operations@iohk.io
